chall11/jbaringo.c

```diff
@@ -0,0 +1,137 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+int check_num(int j, int i, const char *code)
+{
+    if (i > 0 && code[i -1] != '\n')
+        return (0);
+    while (isdigit(code[j]))
+    {
+        if (code[j] != code[i])
+            return (0);
+        j++;
+        i++;
+    }
+    return (i);
+}
+
+char *ft_strjoin(char *s1, char c)
+{
+    char* result;
+    int i = 0;
+
+    if(!(result = malloc(sizeof(char) * strlen(s1) + 1 + 1)))
+        return (NULL);
+    while (s1[i])
+    {
+        result[i] = s1[i];
+        i++;
+    }
+    result[i] = c;
+    result[i+1] = '\0';
+    return (result);
+}
+
+int *goto_reminder(int j, int *loop)
+{
+    int i = 0;
+
+    while (loop[i])
+    {
+        if (loop[i] == j)
+            return (NULL);
+        i++;
+    }
+    int *loop2;
+
+    if (!(loop2 = malloc (sizeof(int) * i + 1)))
+            return (NULL);
+    i = 0;
+    while (loop[i])
+    {
+        loop2[i] = loop[i];
+        i++;
+    }
+    loop2[i] = j;
+    return (loop2);
+}
+
+int check_goto(int i, const char *code)
+{
+    while (code[i] != '\n')
+    {
+        if (code[i] == 'g' && code[i+1] == 'o' && code[i+2] == 't' && code[i+3] == 'o')
+        {
+            while (!isdigit(code[i]))
+                i++;
+            return (i);
+        }
+        i++;
+    }
+    return (0);
+}
+
+char *ft_goto_parser(const char *code)
+{
+    char *result_str;
+    int i = 0;
+    int j = 0;
+    int aux;
+    int space = 1;
+    int *loop;
+
+    if (!(loop = malloc (sizeof(int) * 1)))
+            return (NULL);
+    loop[0] = 0;
+    result_str = strdup("");
+    while (code[i])
+    {
+        if ((j = check_goto(i, code)) != 0)
+        {
+            if ((loop = goto_reminder(j, loop)) == NULL)
+                 return (strdup("Infinite loop !"));
+            i = 0;
+            while (isdigit(code[j]) && code[i])
+            {
+                if (code[j] == code[i])
+                {
+                    if ((aux = check_num(j, i, code)) != 0)
+                    {
+                        i = aux;
+                        j = 0;
+                        break;

```

Comment:
I think `break` statements are confusing, could you reformulate this loop without it? There's always a way ;)